NFS: Revamp example, add nfs-common to debian saltbase

zmerlynn opened the pull request:
This ensures nfs-common is installed on GCE, and provides a more
functional explanation/example. I launched two pods so that there was
a busybox pod to poke around at the NFS volume, and so that the later
wget actually works (the original example would have to work on the
node, or need some other access to the container network). After
switching to two pods, it actually makes more sense to use PV claims,
and it's probably a configuration that makes more sense for
indirection for NFS anyways.

zmerlynn committed:
NFS: Revamp example, add nfs-common to debian saltbase

This ensures nfs-common is installed on GCE, and provides a more
functional explanation/example. I launched two replication controllers
so that there were busybox pods to poke around at the NFS volume, and
so that the later wget actually works (the original example would have
to work on the node, or need some other access to the container
network). After switching to two controllers, it actually makes more
sense to use PV claims, and it's probably a configuration that makes
more sense for indirection for NFS anyways.

zmerlynn commented:
cc @thockin @jsafrane

zmerlynn commented:
@k8s-bot test this

zmerlynn commented:
See #15995 for an explanation of the hardcoded IP.

bgrant0607 commented on examples/nfs/nfs-busybox.yaml:
```diff
@@ -0,0 +1,27 @@
+#
+
+apiVersion: v1
+kind: Pod
```
Why not a RC?

bgrant0607 commented on examples/nfs/nfs-web-pod.yaml:
```diff
@@ -1,11 +1,13 @@
 #
 
 apiVersion: v1
 kind: Pod
```
Why not a RC?

zmerlynn replied:
Will convert to RCs while I'm here. It's fast enough.

bgrant0607 commented on examples/nfs/README.md:
````diff
@@ -33,57 +33,105 @@ Documentation for other releases can be found at
+[NFS service](nfs-server-service.yaml):
+
+```console
+$ kubectl create -f examples/nfs/nfs-server-pod.yaml
````
Why not a RC?

jsafrane commented:
I know it's not part of your change, but would you consider using gcr.io/google_containers/volume-nfs as NFS server? It's from test/images/volumes-tester/nfs/ and I hope it will be tested by e2e tests more often than my private jsafrane/nfs-data image.

zmerlynn commented:
@jsafrane: I absolutely would, but I just spent an hour or so futzing with the official image and while your private image seems to work fine in the example, the official image seems to be broken, which is actually kind of disturbing. I'm going to put this PR up with the private image and then try to iterate on what's going on there, because it *looks* like it should be a drop-in replacement.

zmerlynn commented:
PTAL, I believe this is ready. I switched everything over to ReplicationControllers. As mentioned above, I attempts to move to `gcr.io/google_containers/volume-nfs` and will continue iterating on that separately, but I think that can stand separately.

jsafrane commented:
LGTM

zmerlynn committed:
Fix volumes/test/nfs: Needs to run insecure to allow container connections

Since this is a container service port anyways, "insecure" is a bit of
a red herring. There's no real security relevance to the incoming port
numbers for the NFS server pod.

This lets us get rid of the examples/nfs/exporter Docker build
(@jsafrane's personal image).

zmerlynn commented:
@jsafrane: I figured out what was going on with the `gcr.io/google_containers/volume-nfs` image and fixed it. I'm not sure how storage e2es would be working, but `insecure` is necessary on my configuration, and was enabled on the `examples` server image but not the test server image. The incoming port number is almost entirely irrelevant (the "security" provided by the privileged port number range for NFS has been laughable for .. forever), and in the container environment it's almost certainly going to be true.

jsafrane commented:
@zmerlynn: good catch, `insecure` was (wrongly) removed by my commit ebcc558734d92f85c9d50095b83474cb42439141

zmerlynn commented:
@thockin: Know you're busy, but ping. :)

thockin commented on examples/nfs/README.md:
````diff
@@ -33,57 +33,115 @@ Documentation for other releases can be found at
+```
+
+You should see output similar to the above if everything is working well. If
+it's not, make sure you changed the invalid IP in the [nfs PV](nfs-pv.yaml) file
````
s/nfs/NFS

thockin commented on examples/nfs/README.md:
````diff
@@ -33,57 +33,115 @@ Documentation for other releases can be found at
+$ kubectl create -f examples/nfs/nfs-web-rc.yaml
+```
+
+This creates two pods, each of which server the `index.html` from above. We can
````
s/server/serve

thockin commented on test/images/volumes-tester/nfs/Makefile:
```diff
@@ -1,6 +1,6 @@
 all: push
 
-TAG = 0.3
+TAG = 0.4
```
already pushed?

zmerlynn replied:
Nope, was waiting for LGTM (will push now).

thockin commented:
nits: feel free to self-lgtm

zmerlynn commented:
Nits addressed. Thanks for the review!

zmerlynn commented:
@k8s-bot unit test this please

zmerlynn commented:
@k8s-bot unit test this please

zmerlynn commented:
@k8s-bot unit test this please
